streaming/streaming.go

```diff
@@ -6,29 +6,32 @@ import (
 	"crypto/md5"
 	"crypto/sha256"
 	"encoding/hex"
+	"encoding/json"
 	"errors"
 	"fmt"
-	"github.com/buger/jsonparser"
-	"github.com/cheggaaa/pb/v3"
-	"github.com/elixir-oslo/crypt4gh/model/headers"
-	"github.com/elixir-oslo/lega-commander/conf"
-	"github.com/elixir-oslo/lega-commander/files"
-	"github.com/elixir-oslo/lega-commander/requests"
-	"github.com/elixir-oslo/lega-commander/resuming"
-	"github.com/logrusorgru/aurora"
 	"io"
 	"io/ioutil"
 	"net/http"
 	"net/url"
 	"os"
 	"path/filepath"
 	"strconv"
+
+	"github.com/buger/jsonparser"
+	"github.com/cheggaaa/pb/v3"
+	"github.com/elixir-oslo/crypt4gh/model/headers"
+	"github.com/elixir-oslo/lega-commander/conf"
+	"github.com/elixir-oslo/lega-commander/files"
+	"github.com/elixir-oslo/lega-commander/requests"
+	"github.com/elixir-oslo/lega-commander/resuming"
+	"github.com/golang-jwt/jwt"
+	aurora "github.com/logrusorgru/aurora/v3"
 )
 
 // Streamer interface provides methods for uploading and downloading files from LocalEGA instance.
 type Streamer interface {
-	Upload(path string, resume bool) error
-	uploadFolder(folder *os.File, resume bool) error
+	Upload(path string, resume bool, straight bool) error
```

Comment:
I feel the use of the word `straight` does not reflect the correct meaning. The branch is called direct upload, so why bother to deviate from that?

Reply:
@a-ghanem 

The reason is that
The flags for command of this command-line code that we write it(here is lega commander) should be distinct as possible. If I want to use `direct` then I need to use `-d` as the short form of it. which might be confused with users for other flags of `-d` for delete for the inbox command. This is the reason why I prefere `s` over `d` for this new flag. Users might confuse and delete the files instead of uploading them directly!

Reply:
We agreed to use the "beta" keyword instead, as it does not introduce a conflict, and it will only be used by developers mostly for test purposes until the functionality is stable to become the default behaviour (and not needing a switch/flag of it's own)